### DISCUSSION TITLE
Ignore specific dependencies

### DISCUSSION THREAD

**Discussion by tristanlins** [ben-manes/gradle-versions-plugin#867] [Ideas]
In some situations, dependency updates are irrelevant. For example, when developing a Gradle plugin and using the `kotlin-dsl` plugin, because the internal Gradle Kotlin version is used in that case. 

Currently, however, all Kotlin updates are reported, even though you cannot change the version.

You can solve this with `rejectVersionIf`, for example, with the following condition:

```kotlin
(candidate.module.startsWith("org.gradle.kotlin.kotlin") ||
            candidate.group.startsWith("org.jetbrains.kotlin")) &&
            currentVersion != candidate.version
```

I don't know if it's possible to automatically determine that the internal Kotlin version should be used, but a filter could simplify this:

```kotlin
tasks.withType<DependencyUpdatesTask> {
    // if module is ignored, it will not be checked for updates
    ignoreModuleIf {
        candidate.module.startsWith("org.gradle.kotlin.kotlin") ||
            candidate.group.startsWith("org.jetbrains.kotlin")
    }

    // any ignored module will not be checked here
    ignoreVersionIf { ... }
}
```

Unfortunately, `checkConstraints` and `checkBuildEnvironmentConstraints` do not help here either.

Any suggestions?

**Comment by ben-manes** [comment]
It’s not the prettiest, but the most flexible approach is to use a custom report format to modify the results before printing them using our built-in ones.

Here is an [example](https://github.com/ben-manes/caffeine/blob/0bedba12102dd30c130080f0bf975d5b5ed02094/gradle/plugins/src/main/kotlin/lifecycle/dependency-versions-caffeine-conventions.gradle.kts#L23) when I did this while a dependency had bad metadata on central. I moved it into the correct section until they got Sonatype’s help and fixed that. You could easily remove or otherwise adjust similarly.

**Reply by ben-manes** [reply]
Oh! And I do [exactly](https://github.com/ben-manes/caffeine/blob/b4cedbc411130b8e78c51effdd527756bdf1ff55/gradle/plugins/build.gradle.kts#L64) your workaround for the kotlin dsl or [force versions](https://github.com/ben-manes/caffeine/blob/b4cedbc411130b8e78c51effdd527756bdf1ff55/gradle/plugins/src/main/kotlin/lifecycle/dependency-versions-caffeine-conventions.gradle.kts#L22) when newer are incompatible with my target jdk

**Comment by tristanlins** [comment]
I'm fine with the workaround as well. I thought maybe we could create a convenience method to make it easier.

Or even automatically determine that certain dependency reports are unnecessary. However, I currently don't see how that could work. Even though I have built one or two Gradle plugins, I'm not familiar with the details in the area of configurations 🙈.
